src/Chapter2.hs

```diff
@@ -842,7 +872,8 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate x y = if x < 0 then [] else take (length y) (drop x (cycle y))
```

Comment:
That's a correct solution! Though, it does a little bit more work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face:

Reply:
yes, I can tak

`(drop (mode x y) (cycle y))`

instead of 

`(drop (mode x y) (cycle y)) `